src/commands/init/ai-rules.ts

```diff
@@ -0,0 +1,447 @@
+  )
+  return match ? match.key : null
+}
+
+// Get command and parent PID (same logic as ai-context)
+const getCommandAndParentPID = async (
+  pid: number,
+): Promise<{
+  parentPID: number
+  command: string
+  consumerKey: string | null
+}> => {
+  const { stdout } = await execa('ps', ['-p', String(pid), '-o', 'ppid=,comm='])
+  const output = stdout.trim()
+  const spaceIndex = output.indexOf(' ')
+  const parentPID = output.substring(0, spaceIndex)
+  const command = output.substring(spaceIndex + 1).toLowerCase()
+
+  const consumers = await getContextConsumers(version) // Use current CLI version
+
+  return {
+    parentPID: Number(parentPID),
+    command,
+    consumerKey: getConsumerKeyFromCommand(command, consumers),
+  }
+}
+
+// Detect IDE by walking up process tree (same logic as ai-context)
+const detectIDE = async (): Promise<ConsumerConfig | null> => {
+  if (process.env.AI_CONTEXT_SKIP_DETECTION === 'true') {
+    return null
+  }
+
+  const ppid = process.ppid
+  let result: Awaited<ReturnType<typeof getCommandAndParentPID>>
+  try {
+    result = await getCommandAndParentPID(ppid)
+    while (result.parentPID !== 1 && !result.consumerKey) {
+      result = await getCommandAndParentPID(result.parentPID)
+    }
+  } catch {
+    // Process detection failed
+    return null
+  }
+
+  if (result.consumerKey) {
+    const consumers = await getContextConsumers(version)
+    const contextConsumer = consumers.find((consumer) => consumer.key === result.consumerKey)
+    if (contextConsumer) {
+      return contextConsumer
+    }
+  }
+
+  return null
+}
+
+// Generate MCP configuration for the detected IDE
+const generateMcpConfig = (ide: ConsumerConfig): string => {
+  const configs: Record<string, string> = {
+    vscode: JSON.stringify(
+      {
+        servers: {
+          netlify: {
+            type: 'stdio',
+            command: 'npx',
+            args: ['-y', '@netlify/mcp'],
+          },
+        },
+      },
+      null,
+      2,
+    ),
+    cursor: JSON.stringify(
+      {
+        mcpServers: {
+          netlify: {
+            command: 'npx',
+            args: ['-y', '@netlify/mcp'],
+          },
+        },
+      },
+      null,
+      2,
+    ),
+    windsurf: JSON.stringify(
+      {
+        mcpServers: {
+          netlify: {
+            command: 'npx',
+            args: ['-y', '@netlify/mcp'],
+          },
+        },
+      },
+      null,
+      2,
+    ),
+  }
+
+  return (
+    configs[ide.key] ||
+    JSON.stringify(
+      {
+        mcpServers: {
+          netlify: {
+            command: 'npx',
+            args: ['-y', '@netlify/mcp'],
+          },
+        },
+      },
+      null,
+      2,
+    )
+  )
+}
+
+// VS Code specific MCP configuration
+const configureMcpForVSCode = async (config: string, projectPath: string): Promise<void> => {
+  const configPath = resolve(projectPath, '.vscode', 'mcp.json')
```

Comment:
nit: Save the path to the `.vscode` directory in a variable to avoid resolving the path multiple times.